### PR TITLE
 [BUG][Python] Some regex patterns can generate invalid code. #6675 

### PR DESCRIPTION
**Pull request by kmc026**
This commit is attempting to fix the issue #6675.
According to the issue, when specific regex patterns are used in the OpenAPI declaration yaml files, using Python will cause a formatting issue where a backslash \ will be added after an opening bracket [ and cause a double backslash \ in the outputted sample Default API file. The validation for the sample Default API will then fail.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
@kmc026 thanks for the PR.

cc @taxpon (2017/07) @frol (2017/07) @mbohlool (2017/07) @cbornet (2017/09) @kenjones-cisco (2017/11) @tomplus (2018/10) @Jyhess (2019/01) @arun-nalla (2019/11) @spacether (2019/11)

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonLegacyClientCodegen.java [hunk 1]
```diff
@@ -333,11 +334,45 @@ public void postProcessPattern(String pattern, Map<String, Object> vendorExtensi
...
+    /**
+     * // CS427 Issue link: https://github.com/OpenAPITools/openapi-generator/issues/6675
+     *
+     * Fixes errors caused by adding backslash to some regex patterns
+     *
+     * @parem pattern Regex pattern defined in the OpenAPI specification file
+     * @parem vendorExtensions HashMap which stores strings used in the generated code templates
+     * @return resultPattern Regex pattern which is or is not transformed
+     */
+    public String fixRegexBackSlashError(String pattern, Map<String, Object> vendorExtensions) {
```
Comment (spacether) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonLegacyClientCodegen.java:350]: How about setVendorExtensionRegex?
Yes it fixes a bug but more importantly it sets the vendor extension regex

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/python/PythonLegacyClientCodegenTest.java [hunk 2]
```diff
@@ -91,9 +93,42 @@ public void testRegularExpressionOpenAPISchemaVersion3() {
...
+        final String path = "/ping";
+        final Operation p = openAPI.getPaths().get(path).getGet();
+        final CodegenOperation op = codegen.fromOperation(path, "get", p, null);
+        Map<String, Object> dummyMap = new HashMap<String, Object>();;
+
+        // Issue #6675: Transform issue_6675_fine_pattern from issue_1517.yaml
+        // Original: ^['\"\-\w\s]+$ -> Transformed: ^[\'\"\-\w\s]+$
+        // Transformation occurs because original regex pattern works fine according to issue
+        Assert.assertEquals(codegen.fixRegexBackSlashError(op.allParams.get(6).pattern, dummyMap), "^[\\'\\\"\\-\\w\\s]+$");
+        // Issue #6675: Transform issue_6675_problematic_pattern_1 from issue_1517.yaml
```
Comment (spacether) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/python/PythonLegacyClientCodegenTest.java:119]: How about testing that the x-regex vendor extension contains the right value?
hat's what our templates need.

**Review by spacether — CHANGES_REQUESTED** [review]
Hi there. Thank you for your PR.
In addition to addressing the comments that I left, can you also remove you samples/ additions? If you want to add testing, you can add a sample spec file like we have in our java layer testing in PythonLegacyClientCodegenTest. If you want to see the samples updated, you can add a new regex in the current spec file that our python sample uses at: `bin/configs/python-legacy.yaml` and if you want to add a test in python you can add it in the `samples/openapi3/client/petstore/python-legacy/tests_manual`

**Comment by spacether** [comment]
@kmc026 can you try the 6.2.1 release?
There was a fix for regex handling in that python client. If that works, can this PR be closed?

**Comment by spacether** [comment]
It looks like this is no longer needed as it was fixed in https://github.com/OpenAPITools/openapi-generator/pull/10920

**Pull request closed without merging**